x-pack/plugins/stack_connectors/server/connector_types/inference/index.ts

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import {
+  SubActionConnectorType,
+  ValidatorType,
+} from '@kbn/actions-plugin/server/sub_action_framework/types';
+import {
+  GenerativeAIForSearchPlaygroundConnectorFeatureId,
+  GenerativeAIForSecurityConnectorFeatureId,
+} from '@kbn/actions-plugin/common';
+import { ValidatorServices } from '@kbn/actions-plugin/server/types';
+import { GenerativeAIForObservabilityConnectorFeatureId } from '@kbn/actions-plugin/common/connector_feature_config';
+import {
+  INFERENCE_CONNECTOR_TITLE,
+  INFERENCE_CONNECTOR_ID,
+} from '../../../common/inference/constants';
+import { ConfigSchema, SecretsSchema } from '../../../common/inference/schema';
+import { Config, Secrets } from '../../../common/inference/types';
+import { InferenceConnector } from './inference';
+import { unflattenObject } from '../lib/unflatten_object';
+
+export const getConnectorType = (): SubActionConnectorType<Config, Secrets> => ({
+  id: INFERENCE_CONNECTOR_ID,
+  name: INFERENCE_CONNECTOR_TITLE,
+  getService: (params) => new InferenceConnector(params),
+  schema: {
+    config: ConfigSchema,
+    secrets: SecretsSchema,
+  },
+  validators: [{ type: ValidatorType.CONFIG, validator: configValidator }],
+  supportedFeatureIds: [
+    GenerativeAIForSecurityConnectorFeatureId,
+    GenerativeAIForSearchPlaygroundConnectorFeatureId,
+    GenerativeAIForObservabilityConnectorFeatureId,
+  ],
+  minimumLicenseRequired: 'enterprise' as const,
+  preSaveEventHandler: async ({ config, secrets, logger, scopedClusterClient, isUpdate }) => {
+    const esClient = scopedClusterClient?.asCurrentUser;
```

Comment:
I'm curious if you really want to use `asCurrentUser` here.  Seems like that will involve ensuring users have access to the `/_inference` API.  Alternatively, `asInternalUser` (or whatever) would bypass that requirement.

Reply:
Great catch!